Sources/AppStoreConnectCLI/Services/Operations/CreateProfileOperation.swift

```diff
@@ -32,5 +32,13 @@ struct CreateProfileOperation: APIOperation {
             .map { $0.data }
             .eraseToAnyPublisher()
     }
+    
+    
 
 }
+
+fileprivate extension ProfileType {
+    var areDeviceIdsRequired: Bool {
+        return !self.rawValue.contains("_APP_STORE")
+    }
+}
```

Comment:
This check might be more obvious if it was in the initializer rather than a private extension.
Also the rawValue comparison requires knowledge of how the profile type is formed. For a reader of the code checking against enums might feel less awkward.

I'm thinking something like:
```swift
        let shouldOmitDeviceIds = [.macAppStore, .iOSAppStore, .tvOSAppStore]
            .contains(options.profileType)

        endpoint = APIEndpoint.create(
            profileWithId: options.bundleId,
            name: options.name,
            profileType: options.profileType,
            certificateIds: options.certificateIds,
            deviceIds: shouldOmitDeviceIds ? options.deviceIds : []
        )
```

What do you think?

Reply:
Thanks, this is much better!